modules/core/src/test/java/org/apache/ignite/cdc/ChangeDataCaptureOrderTest.java

```diff
@@ -0,0 +1,250 @@
+import static org.apache.ignite.cluster.ClusterState.ACTIVE;
+import static org.apache.ignite.configuration.WALMode.FSYNC;
+import static org.apache.ignite.testframework.GridTestUtils.runAsync;
+import static org.apache.ignite.testframework.GridTestUtils.waitForCondition;
+
+/** */
+public class ChangeDataCaptureOrderTest extends GridCommonAbstractTest {
+    /** */
+    public static final String FOR_OTHER_DC = "for-other-dc";
+
+    /** */
+    public static final byte OTHER_DC_ID = 2;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        int segmentSz = 10 * 1024 * 1024;
+
+        cfg.setDataStorageConfiguration(new DataStorageConfiguration()
+            .setCdcEnabled(true)
+            .setWalMode(FSYNC)
+            .setMaxWalArchiveSize(10 * segmentSz)
+            .setWalSegmentSize(segmentSz)
+            .setWalForceArchiveTimeout(WAL_ARCHIVE_TIMEOUT)
+            .setDefaultDataRegionConfiguration(new DataRegionConfiguration()
+                .setPersistenceEnabled(true)));
+
+        cfg.setPluginProviders(new AbstractTestPluginProvider() {
+            @Override public String name() {
+                return "ConflictResolverProvider";
+            }
+
+            @Override public CachePluginProvider createCacheProvider(CachePluginContext ctx) {
+                if (!ctx.igniteCacheConfiguration().getName().equals(FOR_OTHER_DC))
+                    return null;
+
+                return new AbstractCachePluginProvider() {
+                    @Override public @Nullable Object createComponent(Class cls) {
+                        if (cls != CacheConflictResolutionManager.class)
+                            return null;
+
+                        return new TestCacheConflictResolutionManager();
+                    }
+                };
+            }
+        });
+
+        return cfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        stopAllGrids();
+
+        cleanPersistenceDir();
+
+        super.beforeTest();
+    }
+
+    /** Simplest CDC test with usage of {@link IgniteInternalCache#putAllConflict(Map)}. */
+    @Test
+    public void testReadAllKeysWithOtherDc() throws Exception {
+        IgniteConfiguration cfg = getConfiguration("ignite-conflict-resolver");
+
+        IgniteEx ign = startGrid(cfg);
+
+        byte drId = (byte)1;
+
+        ign.context().cache().context().versions().dataCenterId(drId);
+
+        ign.cluster().state(ACTIVE);
+
+        TestCDCConsumer cnsmr = new TestCDCConsumer();
+
+        ChangeDataCapture cdc = new ChangeDataCapture(cfg, null, cdcConfig(cnsmr));
+
+        IgniteCache<Integer, User> cache = ign.getOrCreateCache(FOR_OTHER_DC);
+
+        cnsmr.drId = 1;
+        cnsmr.otherDrId = OTHER_DC_ID;
```

Comment:
Naming should be consistent across the project and tests.
dc or dr? need to choose one and apply everywhere.